python/pyspark/sql/column.py

```diff
@@ -71,6 +71,18 @@ def _to_java_column(col: "ColumnOrName") -> JavaObject:
     return jcol
 
 
+def _to_java_expr(col: "ColumnOrName") -> JavaObject:
+    if isinstance(col, (Column, str)):
```

Comment:
wouldn't this disable `str` as a value? We should probably just allow `Column` only. How does it work in Scala side?

Reply:
On the Scala side, you need to pass an `Expression`:
```
scala> val df = spark.range(1024)
df: org.apache.spark.sql.Dataset[Long] = [id: bigint]

scala> df.hint("rebalance", $"id".expr)
res0: org.apache.spark.sql.Dataset[Long] = [id: bigint]
```

Reply:
As is, the only allowed value for the column hint parameters is `str`: https://github.com/apache/spark/blob/master/python/pyspark/sql/dataframe.py#L971
It probably makes sense to allow `Column` objects as well, as both are supported by the `repartition` and `repartitionByRange` methods.